File: ssh/src/dao/interf/IDao.java
package dao.interf;

import java.io.Serializable;
import java.util.List;

public interface IDao<PK extends Serializable, T>
{
	void save(T entity);

	void persist(T entity);

	void delete(T entity);

	void deleteById(PK id);

	void merge(T entity);

	void saveOrUpdate(T entity);

	void update(T entity);

	T getById(PK id);

	List<T> getAll();
}

File: ash/ash3/Dao/IDao.cs
using System;
using System.Linq;
using System.Text;
using model;
using System.Collections.Generic;

namespace ash3.Dao
{
    public interface IDao<TEntity, TId>
    {
        TEntity getById(TId id);

        void save(TEntity entity);

        void update(TEntity entity);

        void delete(TEntity entity);

        IList<TEntity> getAll();
    }
}
